plugins/modules/foreman_host.py

```diff
@@ -74,74 +74,43 @@
 
 RETURN = ''' # '''
 
-try:
-    from nailgun.entities import (
-        Host,
-        HostGroup,
-    )
-
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_host,
-        find_hostgroup,
-        find_location,
-        find_organization,
-        ForemanEntityAnsibleModule,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
-except ImportError:
-    pass
-
-
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'enabled': 'enabled',
-    'hostgroup': 'hostgroup',
-    'location': 'location',
-    'organization': 'organization',
-}
+from ansible.module_utils.foreman_helper import (
+    ForemanEntityAnsibleModule,
+    parameter_entity_spec
+)
 
 
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
+        entity_spec=dict(
             name=dict(required=True),
-            hostgroup=dict(),
-            location=dict(),
-            organization=dict(),
+            hostgroup=dict(type='entity', required=True, flat_name='hostgroup_id'),
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
             enabled=dict(default='true', type='bool'),
```

Comment:
Did you drop the "present with defaults" state intentionally? I think the host had no parameters yet where this was useful, just checking.

Reply:
Yes, I thought the same. Could it maybe be used to drop the hostgroup from a host?

Reply:
dunno. that's @Fobhep's code :)